UPtables/app/models/course_component.rb

```diff
@@ -0,0 +1,19 @@
+class CourseComponent < ActiveRecord::Base
+  
+  TYPES = ['Vorlesung', 'Übung', 'Seminar', 'Projekt']
```

Comment:
If you want to use enumerated types in ruby I would refer to [stackoverflow](http://stackoverflow.com/questions/75759/enums-in-ruby). In this case I would prefer symbols instead of literal strings.